### ISSUE TITLE
CloudFormation output sync v1.0@alpha (for CI / teamwork)

### ISSUE THREAD

**Issue by johncmckim** [serverless/serverless#1688]
Feature Request Template ------------------------------------------

##### Feature Request:
CloudFormation outputs should be synced to serverless.env.yaml before a deployment. This will ensure that the variables required to deploy a project will always exist before it is deployed.

I would also tentatively suggest a separation between generated environment variables and custom environment variables, i.e. CloudFormation outputs vs manually configured variables. These should exist in separate files to make syncing easier and allow the generated outputs to be gitignored.

This is sort of related to #1468. However, #1468 is only related to displaying outputs.

##### Benefits:
* Better integration with CI/CD processes - i.e. a project can be deployed via CI and developers don't have to manually copy CloudFormation outputs to their repo after a deployment

**With separation of generated variables**
* Ability to gitignore generated outputs
* Ensure env variables set by developers are not overwritten by sync process
* Developers can easily differentiate between generated and custom variables

##### Drawbacks:
**Without separate of generated variables**
* Risk of overwriting manually set variables

##### Additional Details:

Discussion template  ----------------------------------------------

##### TLDR:
Serverless v1.0 alpha needs some kind of variable sync.

##### Discussion:
Deploying Serverless 0.5 was a pain, multiple commands + metasync was difficult to use. Now that it's all in with CloudFormation it is much easier to deploy. However, as I see it currently, meta sync is still an issue when using CI/CD.

I have been testing alpha CI/CD processes in my [serverless-cd-example repo](https://github.com/johncmckim/serverless-cd-example). I have deployed the project only using Travis. After sorting out IAM permissions (trial and error), I finally got a deployment to work. However, the subsequent deployment failed as my repo did not contain the ARN's for the newly created `iamRoleArnLambda` and `serverlessDeploymentBucketName` variables. These are CloudFormation outputs so I expected it to just work rather than having to set those manually.

The CloudFormation output variables are different to other variables as they are generated. As such, they should be automatically managed by `sls` rather than the developer(s). Further to this, I think it should be separate from other manually set environment variables to allow the developer to just focus on managing their custom environment variables rather than the CloudFormation outputs.

I'm open to alternative ideas for this.

**Comment by flomotlik** [comment]
@johncmckim could you post the template here? Generally you should not have to sync any Outputs at all. We're simply updating the Cloudformation template, so the `iamRoleArnLambda` and `serverlessDeploymentBucketName` will simply be used again on the next deployment. If that fails thats a bug

**Comment by johncmckim** [comment]
@flomotlik I just tested this again on the new beta and it works as I had hoped.

When I first tried this, (`1.0.0-alpha.1`), I wasn't able to deploy the project without `iamRoleArnLambda` and `serverlessDeploymentBucketName` set in `serverless.env.yml`.

I just re-tested it on (`1.0.0-beta.1`) and I was able deploy a  project on travis and subsequent deployments worked without `iamRoleArnLambda` and `serverlessDeploymentBucketName`.
